pydantic/_internal/_validate_call.py

```diff
@@ -12,10 +14,56 @@
 from ._config import ConfigWrapper
 from ._namespace_utils import MappingNamespace, NsResolver, ns_for_function
 
+# This should be aligned with `GenerateSchema.match_types`
+#
+# Note: This does not play very well with type checkers. For example,
+# `a: LambdaType = lambda x: x` will raise a type error by Pyright.
+ValidateCallSupportedTypes = Union[
+    LambdaType,
+    FunctionType,
+    MethodType,
+    BuiltinFunctionType,
+    BuiltinMethodType,
+    functools.partial,
+]
+
+VALIDATE_CALL_SUPPORTED_TYPES = get_args(ValidateCallSupportedTypes)
```

Comment:
Why is this necessary? Can we not just use the above?

Reply:
At runtime, we might want to use `isinstance(obj, VALIDATE_CALL_SUPPORTED_TYPES)`